src/http-strategies/Koa.js

```diff
@@ -0,0 +1,220 @@
+ *    find an alternate representation? By default, it does the former. But you
+ *    can set this option to false to have this code just pass on to express.
+ */
+export default class KoaStrategy {
+  constructor(apiController, docsController, options) {
+    const defaultOptions = {
+      tunnel: false,
+      handleContentNegotiation: true
+    };
+
+    this.api = apiController;
+    this.docs = docsController;
+    this.config = Object.assign({}, defaultOptions, options); // apply options
+  }
+
+  // For requests like GET /:type, GET /:type/:id/:relationship,
+  // POST /:type PATCH /:type/:id, PATCH /:type, DELETE /:type/:idOrLabel,
+  // DELETE /:type, GET /:type/:id/links/:relationship,
+  // PATCH /:type/:id/links/:relationship, POST /:type/:id/links/:relationship,
+  // and DELETE /:type/:id/links/:relationship.
+  apiRequest() {
+    const strategy = this;
+    return function *(next){
+      try {
+        const reqObj = yield buildRequestObject(this, strategy.config.tunnel);
+        const resObj = yield strategy.api.handle(reqObj, this);
+        const skip = strategy.sendResources(resObj, this);
+        if(skip){
+          yield next;
+        }
+      }
+      catch (err) {
+        strategy.sendError(err, this);
+      }
+    };
+  }
+
+  // For requests for the documentation.
+  docsRequest() {
+    const strategy = this;
+    return function *(next){
+      try {
+        const reqObj = yield buildRequestObject(this, strategy.config.tunnel);
+        const resObj = yield strategy.docs.handle(reqObj, this);
+        const skip = strategy.sendResources(resObj, this);
+        if(skip){
+          yield next;
+        }
+      }
+      catch (err) {
+        strategy.sendError(err, this);
+      }
+    };
+  }
+
+  sendResources(responseObject, ctx) {
+    if(responseObject.headers.vary) {
+      vary(ctx.res, responseObject.headers.vary);
+    }
+
+    if(!responseObject.contentType) {
+      if(this.config.handleContentNegotiation) {
+        ctx.status = 406;
+      }
+      else {
+        return true;
+      }
+    }
+    else {
+      ctx.set("Content-Type", responseObject.contentType);
+      ctx.status = responseObject.status || 200;
+
+      if(responseObject.headers.location) {
+        ctx.set("Location", responseObject.headers.location);
+      }
+
+      if(responseObject.body !== null) {
+        ctx.body = new Buffer(responseObject.body);
+      }
+    }
+  }
+
+  /**
+   * A user of this library may wish to send an error response for an exception
+   * that originated outside of the JSON API Pipeline and that's outside the
+   * main spec's scope (e.g. an authentication error). So, the controller
+   * exposes this method which allows them to do that.
+   *
+   * @param {Error|APIError|Error[]|APIError[]} errors Error or array of errors
+   * @param {Object} ctx Koa's context object
+   */
+  sendError(errors, ctx) {
+    API.responseFromExternalError(errors, ctx.headers.accept).then(
```

Comment:
Will `ctx.headers.accept` here automatically refer to the request's `Accept`? I'm assuming so, right?